src/librustc/lint/builtin.rs

```diff
@@ -95,6 +95,18 @@ declare_lint! {
     "detects overlapping patterns"
 }
 
+declare_lint! {
+    pub BINDING_VARIANT_NAME,
```

Comment:
This does look like a linty lint, but it doesn't match the [lint naming conventions](https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#lints).

Reply:
Hmm... I went back and forth on some names but wasn't aware of the conventions. Would "allow `binding_with_variant_name`" work"? (it's a bit long though...)

Reply:
`binding` -> `bindings`

>it's a bit long though...

I doubt people will have to write it often.

Reply:
Ah, cool. `bindings_with_variant_name` it is then. :)

Reply:
Fixed.